### PR TITLE
Issue/#870

### PR DESCRIPTION
**Pull request by dancgray**
Adds ability to move components within a block without drag and drop. Option only added to component item settings bar.

Also fixes issue #999.

### CODE REVIEW

File: lib/contentmanager.js [hunk 1]
```diff
@@ -1029,6 +1029,38 @@ ContentManager.prototype.setupRoutes = function () {
...
+
+    that.update(type, { _id: switchId }, switchVal, function (error, updatedDocument) {
+      if (error) {
+        res.statusCode = error instanceof ContentTypeError ? 400 : 500;
+        return res.json({ success: false, message: error.message });
+      }
+    });
+
+    res.statusCode = 200;
+    return res.json({ success: true });
```
Comment (dennis-learningpool) [inline lib/contentmanager.js]: lines 1044 to 1060 could result in 3 responses to the client, should the updates fail. Also, it's possible that the first update fails, and the second would succeed, which would be undesirable, I think. 

I think that the update in lines 1052-1057 should be moved inside the first update callback, just after line 1049, and the return on 1059-1060 moved inside that second callback if there is no error.

Also, if there is an error on the second callback, the first update should be rolled back before returning the error.

**Comment by dancgray** [comment]
Thanks @dennis-learningpool, corrected mistake. Added rollback, if that fails it's error is sent to logger.

**Comment by dennis-learningpool** [comment]
Tested and works very nicely +1

**Comment by dennis-learningpool** [comment]
Also, more fun to play with than is really reasonable

**Comment by markl55** [comment]
Dennis gave me a quick preview of this earlier - really nice work Dan.

**Comment by taylortom** [comment]
+1, really *really* cool! :tada: :balloon:

**Comment by dancgray** [comment]
PR is to master, should be to develop. Will recreate for correct branch

**Pull request closed without merging**